### PR TITLE
Beta: record usage of async interface

### PR DESCRIPTION
**Pull request by richardm-stripe**

### CODE REVIEW

File: stripe/_http_client.py [hunk 1]
```diff
@@ -502,7 +502,7 @@ async def _request_with_retries_internal_async(
         *,
         _usage: Optional[List[str]] = None
     ) -> Tuple[Any, int, Mapping[str, str]]:
-        self._add_telemetry_header(headers)
```
Comment (richardm-stripe) [inline stripe/_http_client.py:505]: this was a bug. This meant that telemetry wasn't being recorded at all for async requests

**Review by anniel-stripe — APPROVED** [review]

File: stripe/_http_client.py (hunk 1 above)
Comment (anniel-stripe) [reply stripe/_http_client.py:505]: Nice catch! Looks like there's some tests that need updating after fixing this bug.

Comment (richardm-stripe) [reply stripe/_http_client.py:505]: Fixed in [Fix tests](https://github.com/stripe/stripe-python/pull/1266/commits/45f16ebc99b30b9d07e453601ef5d410c1c845ba)

File: tests/test_integration.py [hunk 2]
```diff
@@ -329,7 +330,7 @@ def do_request(self, req_num):
         assert "last_request_metrics" in telemetry
 
         usage = telemetry["last_request_metrics"]["usage"]
-        assert usage == ["stripe_client"]
+        assert usage == ["stripe_client", "async"]
```
[the remark this answers is not in the document]
Comment (richardm-stripe) [reply tests/test_integration.py:333]: I'm ok with `test_stripe_client_py`, integration tests are a bit slower and don't need to have complete coverage.

**Pull request merged**